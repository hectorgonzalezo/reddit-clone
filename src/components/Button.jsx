import React from 'react';
import { string, bool, func } from 'prop-types';
import styled from 'styled-components';

const StyledButton = styled.button`
  border: none;
  background-color: ${(props) => props.light ? 'white' : 'var(--reddit-blue)'};
  outline: ${(props) => props.light ? '0.1rem solid var(--reddit-blue)' : ''};
  color: ${(props) => props.light ? 'var(--reddit-blue)' : 'white'};
  border-radius: 1rem;
  padding: 4px 13px 4px 13px;
  font-family: inherit;
  font-weight: bold;
  font-size: 0.8rem;
  &:hover{
    background-color: ${(props) => props.light ? 'var(--reddit-blue-super-light)' : 'var(--reddit-blue-light)'};
  }
`;

function Button({ text, light, onClick }) {
  return <StyledButton type="button" onClick={onClick} light={light} className="default-button">{text}</StyledButton>;
}

Button.defaultProps = {
  text: '',
  light: false,
  onClick: () => {},
};

Button.propTypes = {
  text: string,
  light: bool,
  onClick: func,
};

export default Button;
