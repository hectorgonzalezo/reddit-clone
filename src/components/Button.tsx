import React, { SyntheticEvent } from 'react';
import styled from 'styled-components';

interface ButtonProps {
  text?: string;
  data?: string | null;
  light?: boolean;
  type?: "button" | "submit" | "reset";
  disabled?: boolean;
  children?: React.ReactNode;
  onClick: (e: SyntheticEvent) => void;
  onMouseEnter?: () => void;
  onMouseLeave?: () => void;
};

const StyledButton = styled.button<ButtonProps>`
  background-color: ${(props) => props.light ? 'white' : 'var(--reddit-blue)'};
  border: ${(props) => props.light ? '0.1rem solid var(--reddit-blue)' : 'none'};
  color: ${(props) => props.light ? 'var(--reddit-blue)' : 'white'};
  -webkit-border-radius: 1rem;
    -moz-border-radius: 1rem;
  border-radius: 1rem;
  position: relative;
  overflow: hidden;
  padding: 6px 13px 6px 13px;
  font-family: inherit;
  font-weight: bold;
  font-size: 0.8rem;
  &:hover:not(:disabled){
    background-color: ${(props) => props.light ? 'var(--reddit-blue-super-light)' : 'var(--reddit-blue-light)'};
  }

  &:disabled{
    opacity: 0.5;
  }
`;

function Button({
  text = '',
  data = '',
  light= false,
  type= "button",
  disabled= false,
  children= [],
  onClick= (e: SyntheticEvent) => {},
  onMouseEnter= () => {},
  onMouseLeave= () => {},
}: ButtonProps): JSX.Element {
  return (
    <StyledButton
      type={type}
      light={light}
      className="default-button"
      disabled={disabled}
      data={data}
      onClick={onClick}
      onMouseEnter={onMouseEnter}
      onMouseLeave={onMouseLeave}
      value={text}
    >
      {text}
      {children}
    </StyledButton>
  );
  };


export default Button;
