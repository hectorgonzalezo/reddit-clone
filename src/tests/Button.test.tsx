import React from 'react';
import { render, screen, container } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import Button from '../components/Button';

describe('Styled button works', () => {
  test('Renders text', () => {
    render(<Button text="Text" />);
    expect(screen.getByRole('button', { name: 'Text' })).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
  test('Can make clear button', () => {
    render(<Button light />);
    expect(screen.getByRole('button')).toHaveStyle('background-color: white');

    expect(container).toMatchSnapshot();
  });


  test('Calls onclick function', () => {
    const mockFunc = jest.fn();
    render(<Button onClick={mockFunc} />);
    const button = screen.getByRole('button');
    userEvent.click(button);

    expect(mockFunc).toBeCalled();
  });
})