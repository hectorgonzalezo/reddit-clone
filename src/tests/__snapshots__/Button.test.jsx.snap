// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled button works Can make clear button 1`] = `undefined`;

exports[`Styled button works Renders text 1`] = `undefined`;
